sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -211,39 +211,38 @@ def test_classification_synthetic(loss):
     check_classification_synthetic(loss)
 
 
-def check_boston(loss, subsample):
-    # Check consistency on dataset boston house prices with least squares
+def check_regression_dataset(loss, subsample):
+    # Check consistency on regression dataset with least squares
     # and least absolute deviation.
-    ones = np.ones(len(boston.target))
+    ones = np.ones(len(y_reg))
     last_y_pred = None
-    for sample_weight in None, ones, 2 * ones:
+    for sample_weight in [None, ones, 2 * ones]:
         clf = GradientBoostingRegressor(n_estimators=100,
                                         loss=loss,
                                         max_depth=4,
                                         subsample=subsample,
                                         min_samples_split=2,
                                         random_state=1)
 
-        assert_raises(ValueError, clf.predict, boston.data)
-        clf.fit(boston.data, boston.target,
-                sample_weight=sample_weight)
-        leaves = clf.apply(boston.data)
-        assert leaves.shape == (506, 100)
+        assert_raises(ValueError, clf.predict, X_reg)
+        clf.fit(X_reg, y_reg, sample_weight=sample_weight)
+        leaves = clf.apply(X_reg)
+        assert leaves.shape == (500, 100)
 
-        y_pred = clf.predict(boston.data)
-        mse = mean_squared_error(boston.target, y_pred)
-        assert mse < 6.0
+        y_pred = clf.predict(X_reg)
+        mse = mean_squared_error(y_reg, y_pred)
+        assert mse < 0.04
 
         if last_y_pred is not None:
-            assert_array_almost_equal(last_y_pred, y_pred)
-
+            assert_allclose(last_y_pred, y_pred)
```

Comment:
@lucyleeow I added your test here. Using the strategy `nearest` everywhere seems to be the winner here to be able to keep the `sample_weight` semantic rights.